torch/testing/_comparison.py

```diff
@@ -832,14 +839,14 @@ def _compare_regular_values_close(
     ) -> None:
         """Checks if the values of two tensors are close up to a desired tolerance."""
         actual, expected = self._promote_for_comparison(actual, expected)
-        mismatches = ~torch.isclose(actual, expected, rtol=rtol, atol=atol, equal_nan=equal_nan)
-        if not torch.any(mismatches):
+        matches = torch.isclose(actual, expected, rtol=rtol, atol=atol, equal_nan=equal_nan)
+        if torch.all(matches):
```

Comment:
@mruberry This should resolve the performance degradation that you saw in the internal tests. The old `assertEqual` invoked `torch.allclose`

https://github.com/pytorch/pytorch/blob/26d54b4076fe6b6fd6e6c517643a19cfa5dcd7d9/torch/testing/_core.py#L96-L97

which is comparable to `torch.isclose` and `torch.all` here. 

The slowdown happened because we previously also inverted the array to include some mismatch statistics in the error message. Since we only need the mismatches in the error case, we can safely push the inversion there.